Issue 모델을 기반으로 한 API 구현

GodDrinkTeJAVA opened the pull request:
1. 모든 이슈 읽어오기
2. 이슈 생성
3. 이슈를 issue_id를 기반으로 가져오기
4. issue를 업데이트하기
5. id를 기반으로 issue 삭제하기

GodDrinkTeJAVA committed:
[FEAT] 코멘트 포스트 기능 추가

미들웨어에서 구현만 된 상태인 createComment를 라우트했다.
이제 /issue/:issueid/comment로 POST 요청을 보내면 댓글을 달 수 있다.

GodDrinkTeJAVA committed:
[FEAT] Issue를 모델로 하는 middleware 구현

구현 목록은 다음과 같습니다
1. 모든 이슈 읽어오기
2. 이슈 생성
3. 이슈를 issue_id를 기반으로 가져오기
4. issue를 업데이트하기
5. id를 기반으로 issue 삭제하기

GodDrinkTeJAVA committed:
[MERGE] Merge conflict solved

GodDrinkTeJAVA committed:
[FIX] Issue PUT 요청이 처리되지 않는 버그 수정

해당 값의 userID가 DB 작업 수행 시 반영되지 않았다.
값이 전달되도록 인자에 추가하였다.

GodDrinkTeJAVA committed:
[FEAT] Issue state를 toggle하는 API 개발

원래 false였다면 true로 만들라고 하고, 아니면 false로 바꿉니다.

hyh1016 commented on server/middlewares/issue.js:
```diff
@@ -0,0 +1,83 @@
+      revertedState = 1;
+  }
+  else {
+      revertedState = 0;
```
OPEN을 선언하셨으니 CLOSE=0도 선언해서 적용하시면 좀 더 가독성이 좋아질 것 같습니다!

bbbyung2 commented on server/middlewares/issue.js:
```diff
@@ -41,19 +41,19 @@ const updateIssue = async (req, res) => {
 const deleteIssue = async (req, res) => {
   const { issueid: issueID } = req.params;
-
+    console.log(issueID);
```
이것만 삭제하면 좋을듯 합니다!

hyh1016 commented on server/models/query.js:
```diff
@@ -4,6 +4,14 @@ module.exports = {
   UPDATE_COMMENT: 'update comment set contents = ? where id = ?;',
   DELETE_COMMEN: 'delete from comment where id = ?;',
 
+  READ_ALL_ISSUE: 'select * from issue',
```
*을 사용하면 어떤 attribute가 있는 건지 바로 확인이 어려워서 안 좋다는 이야기를 들었던 것 같아요..!
모든 attribute명을 적어주시는 게 더 좋지 않을까요?
ex) select id, title, contents, is_open, user_id, milestone_id from issue

hyh1016 reviewed (commented):
황금버섯 준희님 👍👍

GodDrinkTeJAVA committed:
[REFACTOR] middlewares/issue 수정

deleteIssue의 불필요한 로그 함수 제거
magic number 제거 및 기존 선언 상수로 대체

GodDrinkTeJAVA committed:
[REFACTOR] middlewares/issue, models/query 수정

deleteIssue의 불필요한 로그 함수 제거
magic number 제거 및 기존 선언 상수로 대체
SQL query에서 * 대신 column 나열식으로 변경

GodDrinkTeJAVA committed:
Merge branch 'develop' of https://github.com/GodDrinkTeJAVA/IssueTracker-29 into develop

GodDrinkTeJAVA committed:
[FIX] issue query에서 milestone_id가 READ_ISSUE_BY_ID에 빠진 오류 수정
